torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp

```diff
@@ -1037,6 +1014,53 @@ void ExprSegmentationSorter::mergeNodes() {
   });
 }
 
+// Initialize concrete_id_dependencies and concrete_id_to_all_ids
+void ExprSegmentationSorter::initializeForLoopDependencies() {
+  TORCH_INTERNAL_ASSERT(
+      concrete_id_dependencies.empty(),
+      "For loop dependencies have already been initialized.");
+
+  for (auto tv : ir_utils::allTvs(complete_fusion_)) {
+    std::unordered_set<IterDomain*> dependencies;
+    for (size_t tv_id_i =
+             std::max(tv->getMaxProducerPosition(), tv->getComputeAtPosition());
+         tv_id_i > 0;
+         tv_id_i--) {
+      auto tv_id = tv->axis(tv_id_i - 1);
+      auto concrete_id =
+          GpuLower::current()->caLoopMap().getConcreteMappedID(tv_id);
+
+      if (concrete_id_dependencies.find(concrete_id) ==
+          concrete_id_dependencies.end()) {
+        concrete_id_dependencies[concrete_id] = dependencies;
+      } else {
+        concrete_id_dependencies[concrete_id].insert(
+            dependencies.begin(), dependencies.end());
+      }
+
+      // Loops after tv_id are dependent on tv_id
+      dependencies.emplace(
+          GpuLower::current()->caLoopMap().getConcreteMappedID(tv_id));
+    }
+  }
+}
+
+// Checks if the for loop associated with the concrete ID is ready to be
+// resolved in sorting. This could be done more efficiently with some
+// additional tracking, however we recreate ca_domain_ when we merge groups,
+// so it's hard to track what is no longer needed.
+bool ExprSegmentationSorter::loopReady(IterDomain* concrete_id) {
+  auto dependencies = concrete_id_dependencies[concrete_id];
+  for (auto& group : groups_) {
+    for (auto ca_domain : group->payload()->ca_domains_) {
```

Comment:
Still trying to understand the logic here. The parameter of this function gets the last domain of `producer_ca_domain` or `consumer_pa_domain`. Here, it only checks `ca_domains_`. Doesn't it need to look at `pa_domains_` as well?